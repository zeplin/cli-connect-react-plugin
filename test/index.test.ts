import Plugin from "../src";

describe("Connected Components React Plugin", () => {
    test("Component.jsx snippet creation", async () => {
        const plugin = new Plugin();

        const componentCode = await plugin.process(
            {
                path: "test/samples/Component.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithChildren.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/ComponentWithChildren.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/ComponentWithProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("ComponentWithChildrenAndProps.jsx snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/ComponentWithChildrenAndProps.jsx",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });
});
