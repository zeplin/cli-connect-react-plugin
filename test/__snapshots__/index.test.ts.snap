// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactLinkProcessor MyComponent.jsx snippet creation 1`] = `
Object {
  "description": "Component description.",
  "lang": "jsx",
  "snippet": "<MyComponent
  optionalArray={array}
  optionalBool={bool}
  optionalFunc={func}
  optionalNumber={number}
  optionalObject={object}
  optionalString={string}
  optionalSymbol={symbol}
  optionalNode={node}
  optionalElement={element}
  optionalElementType={elementType}
  optionalFoo={instanceOf(Foo)}
  optionalEnum={enum}
  optionalUnion={union[string|number|instanceOf(Foo)}]
  optionalArrayOf={arrayOf[number]}
  optionalObjectOf={objectOf[number]}
  optionalObjectWithShape={shape}
  optionalObjectWithStrictShape={exact}
  requiredFunc={func}
  requiredAny={any}
  customProp={custom}
  customArrayProp={arrayOf[custom]}
  customObjectOfProp={objectOf[custom]}>
{children}
</MyComponent>",
}
`;
